overlord/devicestate/devicestate_test.go

```diff
@@ -801,9 +803,17 @@ func (s *deviceMgrSuite) TestDoRequestSerialErrorsOnNoHost(c *C) {
 		c.Skip("cannot run test when http proxy is in use, the error pattern is different")
 	}
 
+	const nonexistent_host = "nowhere.nowhere.test"
+
+	// check internet access
+	_, err := net.LookupHost(nonexistent_host)
+	if err != nil && !strings.Contains(err.Error(), "no such host") {
```

Comment:
Thanks for your work on fixing the tests! I wonder if  instead of the string check we should bail if the error is not a net error or if it is not a permanent net error (basicly mirror the check in devicestate/handlers.go which errors only if `netErr, ok := err.(net.Error); ok && !netErr.Temporary()`. I'm happy to do this change if there is agreement from @niemeyer and @zyga

Reply:
yeah, this is a great idea +1 mvo.